test/framework/src/main/java/org/elasticsearch/index/mapper/FieldTypeTestCase.java

```diff
@@ -20,50 +20,16 @@
 
 import org.elasticsearch.index.query.QueryShardContext;
 import org.elasticsearch.test.ESTestCase;
-import org.elasticsearch.test.EqualsHashCodeTestUtils;
-
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
 
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 /** Base test case for subclasses of MappedFieldType */
-public abstract class FieldTypeTestCase<T extends MappedFieldType> extends ESTestCase {
+public abstract class FieldTypeTestCase extends ESTestCase {
```

Comment:
It might be worth dropping this in a followup. It is just about empty now.

Reply:
The mock shard contexts are still pretty widely used, so I think it's worth keeping for now.